src/lib/operators/table_scan.cpp

```diff
@@ -300,8 +300,8 @@ std::unique_ptr<AbstractTableScanImpl> TableScan::create_impl() const {
 
     auto lower_bound_value = expression_get_value_or_parameter(*between_expression->lower_bound());
     if (lower_bound_value) {
-      const auto adjusted_predicate_and_value =
-          lossless_predicate_variant_cast(lower_condition, *lower_bound_value, left_column->data_type());
+      const auto adjusted_predicate_and_value = lossless_predicate_variant_cast(
+          lower_condition, *lower_bound_value, between_expression->value()->data_type());
       if (adjusted_predicate_and_value) {
         lower_condition = adjusted_predicate_and_value->first;
         lower_bound_value = adjusted_predicate_and_value->second;
```

Comment:
I know BETWEEN statements, but this implementation seemed Greek to me. 

1. Split up predicate:`auto [lower_condition, upper_condition] = between_to_conditions(predicate_condition);`
2. Some modifications -> **please comment what and why**
3. Bringt it back together again: `conditions_to_between(lower_condition, upper_condition)`

Reply:
Good point. Does this help?

```
The ColumnBetweenTableScanImpl expects both values to be of the same data type as the column that is being
scanned. We retrieve the lower and upper bounds of the BetweenExpression, perform a
lossless_predicate_variant_cast into the column's data type and reassemble the between condition.
```

Reply:
> Good point. Does this help?
> 
> ```
> The ColumnBetweenTableScanImpl expects both values to be of the same data type as the column that is being
> scanned. We retrieve the lower and upper bounds of the BetweenExpression, perform a
> lossless_predicate_variant_cast into the column's data type and reassemble the between condition.
> ```

Yes! However, I wonder why we need such a cast?

Reply:
Added

> Many implementations require the comparison values to be of the same column type. If we were to cast the values to the column type, `int_column = 16.25` would turn into `int_column = 16`, which is obviously wrong. As such, we use lossless casts to guarantee safe type conversions. This was introduced by #1550.

Does that help?

Reply:
Perfect!

Reply:
Thanks for bringing that up. That would have easily gotten lost over time.